File: zhanglianzhuang/day3/demo03/src/main/java/com/example/demo/controller/RoleController.java
package com.example.demo.controller;

import com.example.demo.entity.Role;
import com.example.demo.mapper.RoleMapper;
import com.example.demo.mapper.UserRoleMapper;
import io.swagger.annotations.Api;
import io.swagger.annotations.ApiOperation;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.DeleteMapping;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PutMapping;
import org.springframework.web.bind.annotation.RestController;
import springfox.documentation.swagger2.annotations.EnableSwagger2;

import java.util.List;

/**
 *说明：对角色表进行操作
 * 编写人：张连壮
 * 编写时间：2019年12月25日
 * 修改时间
 */
@Api(tags = "Role")
@RestController
@EnableSwagger2
public class RoleController {

	@Autowired
	private RoleMapper roleMapper;
	@Autowired
	private UserRoleMapper userroleMapper;
	@ApiOperation(value = "角色添加")
	@GetMapping("/addrole")
	public int addRole(Role role){
		List<Role> roles=getRolebyname(role);
		if(roles.size()==0){
			return roleMapper.insertRole(role);
		}
		else {
			System.out.println("角色名已存在！");
		}
		return 0;
	}
	@ApiOperation(value = "角色查询")
	@GetMapping("/getrole")
	public List<Role> selectRole(Role role){
		return roleMapper.getRole(role);
	}
	@ApiOperation(value = "角色更新")
	@PutMapping("/updaterole")
	public int updateRole(Role role){
		return roleMapper.updateRole(role);
	}
	@ApiOperation(value = "角色删除")
	@DeleteMapping("/deleterole")
	public int deleteRole(Role role){
		int flag=userroleMapper.getRolebyid(role);
		if (flag==0) {
			return roleMapper.deleteRoler(role);
		}
		else {
			System.out.println("该角色下还有用户!");
		}
		return 0;
	}

	public List<Role> getRolebyname(Role role){
		return roleMapper.getRolebyname(role);
	}

}

File: caiguicheng/day03/work/demo03work/controller/RoleController.kt
package com.example.demo03work.controller

import com.example.demo03work.entity.Role
import com.example.demo03work.entity.RoleNum
import com.example.demo03work.entity.User
import com.example.demo03work.service.RoleService
import com.example.demo03work.service.UserService
import io.swagger.annotations.Api
import io.swagger.annotations.ApiOperation
import org.mybatis.spring.annotation.MapperScan
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.context.annotation.Lazy
import org.springframework.web.bind.annotation.DeleteMapping
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RestController
import springfox.documentation.swagger2.annotations.EnableSwagger2

/**
 *
 */
@Api(tags = ["role"])
@RestController
@EnableSwagger2
class RoleController {

    //使用时候创建
    @Autowired
    @Lazy
    lateinit var roleService: RoleService

    @ApiOperation("角色添加")
    @GetMapping("/addRole")
    fun addRole(role: Role): Int {
        println("id:")
        println(role.id)
        return roleService.insertRole(role)
    }

    @ApiOperation("删除角色")
    @DeleteMapping("/deleteRole")
    fun deleteUser(id: Int): Int {
        return roleService.deleteRole(id)
    }

    /**
     * 根据条件获得用户
     * @return 符合条件的角色列表
     */
    @ApiOperation("查找角色信息")
    @PostMapping("/selectRole")
    fun selectUser(role: Role): List<Role> {
//        println("time")
//        println(role.id)
        return roleService.selectRole(role)
    }

    /**
     * 显示列表
     * @return 角色列表
     */
    @ApiOperation("查找角色信息")
    @PostMapping("/getList")
    fun getRoleNumber(): List<RoleNum> {
        return roleService.getRoleNumber()
    }

    @ApiOperation("修改角色信息")
    @PostMapping("/updateRole")
    fun updateUser(role: Role): Int {
        return roleService.updateRole(role)
    }
}